marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/HubDataSource.java

```diff
@@ -13,17 +13,33 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.marklogic.hub.cloud.aws.glue.Writer;
+package com.marklogic.hub.spark.sql.sources.v2;
 
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.hub.spark.sql.sources.v2.writer.HubDataWriterFactory;
+import org.apache.spark.sql.SaveMode;
 import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.sources.v2.DataSourceOptions;
+import org.apache.spark.sql.sources.v2.WriteSupport;
 import org.apache.spark.sql.sources.v2.writer.DataSourceWriter;
 import org.apache.spark.sql.sources.v2.writer.DataWriterFactory;
 import org.apache.spark.sql.sources.v2.writer.WriterCommitMessage;
 import org.apache.spark.sql.types.StructType;
 
 import java.util.Map;
+import java.util.Optional;
 
-public class MarkLogicWriter implements DataSourceWriter {
+public class HubDataSource extends LoggingObject implements WriteSupport {
+
```

Comment:
HubDataSource should be inside the writer folder because it implements WriteSupport. It cannot be used as a generic class. For Reader we will have to create a new class that implements ReadSupport.

Reply:
I am thinking that since WriteSupport is in the "v2" package and not "v2.writer", we'll eventually have this implement ReadSupport as well. Here's an example of that - https://www.bugdbug.com/post/speed-up-apache-spark-operations-using-a-custom-data-source . 

I think that makes sense conceptually - we have a single DataSource which supports Read and Write. That makes life easier for Ernie, who only has to worry about one DataSource class now.